### PR TITLE
Added support for release notes asciidoc macros (#1082)

### PR DESCRIPTION
**Pull request by daveoconnor**
Related to ticket #1082, this adds support for asciidoc macros to emulate macros from qbk. The `qbk` items below are the originals from the ticket, the `asciidoc` items below are the new ones for asciidocs, and the `html` items below are the output.

qbk: `[phrase library..XX]`
String to use in asciidoc: `phrase::[library,Hi!]`
html: `<span class="library">Hi!</span>`

qbk: `[@/libs/function/ Function]`
asciidoc: `at::/tools/build/doc/html/#_version_5_1_0[B2 version 5.1.0]`
html: `<a href="/tools/boost/boost_1_75_0/libs/algorithm/doc/html/algorithm/">Algorithm Library</a>`

qbk: `[github_pr geometry 1247]`
asciidoc: `github_pr::[geometry,1247]`
html: `<a href="https://github.com/boostorg/geometry/pull/1247">PR#1247</a>`

qbk: `[github geometry 1231]`
aasciidoc: `github_issue::[geometry,1231]`
html: `<a href="https://github.com/boostorg/geometry/issues/1231">#1231</a>`

qbk: `[phrase library..[@/libs/charconv/ Charconv]:]`
asciidoc: `phrase::[library,at::/libs/charconv/[Charconv:]]`
html: `<span class="library"><a href="/libs/charconv/">Charconv:</a></span>`

### CODE REVIEW

**Comment by vinniefalco** [comment]
These use asciidoctor's framework for authoring macros? Or is it just doing some kind of regex on the input file? I thought asciidoctor extensions had to be written in Ruby?

**Comment by daveoconnor** [comment]
Status update: I've redone these in ruby. Will make this just about the integration between the rubygems package when that's done and our servers.

**Comment by rbbeeston** [comment]
So are we to wait on reviews until the rubygems piece is done?

**Comment by daveoconnor** [comment]
Yeah, it should be today.

**Comment by daveoconnor** [comment]
This is taking a little longer than expected. Should be done tomorrow.

Repo is set up, and while I need to make some adjustments we have initial publishing of gems at https://rubygems.org/gems/asciidoctor-boost. Outstanding items I've seen so far:

1. The documentation is rough-ish
2. Links to things like the changelog
3. A github workflow quirk for publishing of gems 
4. Some other items I'm sure I'll see once I give it a final once over

While the package and commands should work, don't judge the surrounding project too harshly just yet, it's still a WIP.

**Comment by daveoconnor** [comment]
This is now ready for review again. 

Project repo with documentation on the asciidoctor macros:  https://github.com/cppalliance/asciidoctor-boost
Ruby gem page: https://rubygems.org/gems/asciidoctor-boost

**Review by rbbeeston — APPROVED** [review]
ok to merge once Greg's issues are addressed

**Pull request merged**
